scopesim/utils.py

```diff
@@ -646,7 +649,8 @@ def get_meta_quantity(meta_dict, name, fallback_unit=""):
 
     """
     if isinstance(meta_dict[name], str) and meta_dict[name].startswith("!"):
-        meta_dict[name] = from_currsys(meta_dict[name])
+        raise ValueError(f"!-strings should be resolved upstream: "
```

Comment:
It seems that we now want to resolve all !-strings upon initialization, but is this really a good idea? There is something to be said to evaluate all !-strings only when they are actually needed.

In particular, would this new scheme mean that it is not possible to easily change the values later? Say the telescope refers to `!ATMO.temperature`, which is then resolved upon initialization to, say, 7˘C. Then the user changes `!ATMO.temperature` to, say, 8. Would the telescope temperature then stay on 7?

I'm asking, because I'm not sure I understand the new design. (Not sure I understand the old design properly either though.)

Practically though, most files refer to other files (which is the point), but those other files might not yet be loaded. E.g. https://github.com/AstarVienna/irdb/blob/dev_master/ELT/LIST_ELT_combined.tbl refers to `!TEL.temperature`. So it is only possible to instantiate `SurfaceList(filename="ELT/LIST_mirrors_ELT.tbl")` if the atmosphere has been instantiated.

There could also be cyclic dependencies, because https://github.com/AstarVienna/irdb/blob/dev_master/Armazones/Armazones.yaml refers to `!INST.pixel_scale`, but https://github.com/AstarVienna/irdb/blob/dev_master/MICADO/MICADO_IMG_wide.yaml which defines `!INST.pixel_scale` refers to  `!ATMO.altitude`.

Or does this actually work? I'm confused.